src/test/change_stream.rs

```diff
@@ -0,0 +1,711 @@
+use std::time::Duration;
+
+use futures::StreamExt;
+
+use crate::{
+    bson::doc,
+    change_stream::document::{ChangeStreamEventSource, OperationType, UpdateDescription},
+    options::{ChangeStreamOptions, FullDocumentType, UpdateOptions},
+    selection_criteria::{ReadPreference, SelectionCriteria},
+    test::{util::drop_collection, TestClient, LOCK},
+    Namespace,
+    RUNTIME,
+};
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn empty_change_stream() {
+    let _guard = LOCK.run_concurrently().await;
+
+    let client = TestClient::new().await;
+    if client.server_version_lt(4, 0) || client.is_standalone() {
+        return;
+    }
+    let pipeline = vec![doc! { "$match": { "x": 3 } }];
+    let options = ChangeStreamOptions::builder()
+        .selection_criteria(SelectionCriteria::ReadPreference(ReadPreference::Primary))
+        .max_await_time(Duration::from_millis(5))
+        .batch_size(10)
+        .build();
+    let mut change_stream = client.watch(pipeline, Some(options)).await.unwrap();
```

Comment:
Since the `watch` helpers take `impl Into<Option<ChangeStreamOptions>>`, there's no need to use `Some` here; `T` implements `Into<Option<T>>` for all `T`.

Reply:
fixed